roles/roles_common_async.js

```diff
@@ -1263,6 +1264,41 @@ Object.assign(Roles, {
     return false
   },
 
+  /**
+   * Check if current user is in at least one of the target roles.
+   * @method isInRoleAsync
+   * @param {String} role Name of role or comma-seperated list of roles.
+   * @param {String} [scope] Optional, name of scope to check.
+   * @return {Promise} `true` if current user is in at least one of the target roles.
+   * @static
+   */
+  isInRoleAsync: async function (role, scope) {
+    const userId = Meteor.userId()
+    const comma = (role || '').indexOf(',')
+    let roles
+
+    if (!userId) return false
+    if (!Match.test(role, String)) return false
+
+    if (comma !== -1) {
+      roles = role.split(',').reduce(function (memo, r) {
+        if (!r) {
+          return memo
+        }
+        memo.push(r)
+        return memo
+      }, [])
+    } else {
+      roles = [role]
+    }
+
+    if (Match.test(scope, String)) {
+      return await this.userIsInRoleAsync(userId, roles, scope)
+    }
+
+    return await this.userIsInRoleAsync(userId, roles)
```

Comment:
Technically you don't need to await before returning and also don't need to make the function async, because it will return a promise anyway.

If you resolve the promise before returning, there will be one more microtask execution than if returning the promise directly:

```js
isInRoleAsync: function (role, scope) {
  // ...

  return this.userIsInRoleAsync(userId, roles) // always returns Promise
}
```

does not affect anything by itself but might be measurable when requests / operations do in the millions haha